### PR TITLE
Add OrePlaintext<64> encoding for Strings

### PR DESCRIPTION
**Pull request by mpalmer**
What it says on the tin.

### CODE REVIEW

File: src/encode.rs [hunk 1]
```diff
@@ -61,6 +63,11 @@ impl From<u64> for OrePlaintext<u64> {
     }
 }
 
+impl From<String> for OrePlaintext<u64> {
```
Comment (coderdan) [inline src/encode.rs:81]: Isn't `Plaintext` defined in terms of `N` where `N` is the number of bytes? Or is this different to the `Plaintext` used in ore.rs?

Comment (coderdan) [reply src/encode.rs:81]: One other consideration here is that there will be multiple ways to do this conversion depending on the goal. For orderable strings we might use a compressed or even lossy conversion. Could these traits be overloaded to do that? Maybe we don't need to worry for now but just flagging it.

File: src/encode.rs [hunk 2]
```diff
@@ -86,6 +93,11 @@ mod tests {
         assert_eq!(0.0f64.partial_cmp(&-0.0f64), Some(Ordering::Equal))
     }
 
+    #[test]
+    fn test_strings_hash_correctly() -> () {
+        assert_eq!(OrePlaintext::<u64>::from(format!("Hello world!")).0, 16562844544668089185);
```
Comment (freshtonic) [inline src/encode.rs:159]: What is the call to `format!` for?

**Review by freshtonic — APPROVED** [review]
Approved (with question)

File: src/encode.rs (hunk 2 above)
Comment (mpalmer) [reply src/encode.rs:159]: Lazy coder's type conversion, from (IIRC) `&'static str` to `String`.

File: src/encode.rs (hunk 1 above)
Comment (mpalmer) [reply src/encode.rs:81]: Yeah, this is the [other plaintext type](https://www.notion.so/cipherstash/ore_encoding-rs-should-use-ore-rs-PlainText-type-b00d3efa25014c2394a1ad71dc1efad0#9160fbd5f1d949e4801d1bb123b456c1) in action.  Presumably ordered strings or whatever else could go into a different destination type, or something.

This code is a straight-up rip from `node-ore-rs`, pulled in here so I don't have to play copy-pasta games in ruby-ore-rs, BTW.

Comment (coderdan) [reply src/encode.rs:81]: Understood.

**Pull request merged**